### PR TITLE
Added Lists API

### PR DESCRIPTION
**Pull request by omenocal**
For more information about the official documentation, visit:
https://developer.amazon.com/docs/custom-skills/access-the-alexa-shopping-and-to-do-lists.html

You can also take a look at the FAQ:
https://developer.amazon.com/docs/custom-skills/list-faq.html

### CODE REVIEW

File: docs/lists.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+    const listMeta = _.find(listsMetadata.lists, { name: listName });
+    let itemInfo;
+    let listInfo;
+
+    if (listMeta) {
+      listInfo = await alexaEvent.lists.getListById(listMeta.listId);
+      itemInfo = _.find(listInfo.items, { value: productName });
+    } else {
+      listInfo = await alexaEvent.lists.createList(listName);
```
Comment (armonge) [inline docs/lists.rst]: could we maybe add this to the helper? Something that `getsOrCreates` a list

File: lib/apis/lists.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+      body,
+      headers: {
+        Authorization: `Bearer ${this.consentToken}`,
+      },
+      json: true, // Automatically parses the JSON string in the response
+    };
+  }
+
+  getDefaultList(listSuffix) {
+    return rp(this.getOptions())
```
Comment (armonge) [inline lib/apis/lists.js]: async /await ?

File: lib/apis/lists.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+  updateItem(listId, itemId, value, status, version) {
+    if (typeof value === 'object') {
+      return rp(this.getOptions(`${listId}/items/${itemId}`, 'PUT', value));
+    }
+
+    return rp(this.getOptions(`${listId}/items/${itemId}`, 'PUT', { value, status, version }));
+  }
+
+  deleteItem(listId, itemId) {
```
Comment (armonge) [inline lib/apis/lists.js]: would actually love documentation for all this methods

File: lib/apis/lists.js (hunk 2 above)
Comment (omenocal) [reply lib/apis/lists.js]: Travis has a NodeJs 6 test that gets broken when it reads async/await words

**Comment by omenocal** [comment]
@armonge done

**Comment by armonge** [comment]
Let's add more documentations for all the public methods the list helper has, something like https://voxa.readthedocs.io/en/dev/statemachine-skill.html

**Pull request merged**